MissionBit/BAMPWebsite
# Connect email form to back end

This PR delivers **[make newsletter sign-up form work](https://app.asana.com/0/1184406596339075/1199924348903137)**.

For a back-end service, I'm leaning towards **[slapform](https://slapform.com/)**, but was also looking at [formspree](https://formspree.io) and [getform](https://getform.io).

## How to test this branch

1. Enter an email into the newsletter sign-up form. **localhost is fine -- the page does not need to be hosted online.**
2. Check the mailbox at https://mailpoof.com/mailbox/bampsite@mailpoof.com to see if it worked.

### Note on redirect
From the thank-you page, you *should* be taken back to `index.html`, but it will depend on how the page was hosted. If you used GitHub pages, you'll end up at `missionbit.github.io`. For a library like `serve` you may get a file listing, depending on how it's configured.

Neither of those scenarios will be a problem on the production host. Also, we have the option of writing some JS to handle this better. For now, I think the fact that you get a thank you and end up back at the original domain is pretty good.

### Why Slapform?

All of the services I looked at are very simple and they all have a (50 submission / month) free plan. But Slapform lets you do AJAX submission, so that makes it more versatile than the others.

Even without the AJAX feature, you can disable the captcha, and the standard thank-you page looked fine to me. Also, I think they're being lenient about the custom redirect since it worked when I tested it.
